python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -1303,12 +1310,84 @@ def _ensure_persisted_execution_plan_snapshot(
 
         return execution_plan_snapshot_id
 
-    def _log_asset_planned_events(
-        self, dagster_run: DagsterRun, execution_plan_snapshot: "ExecutionPlanSnapshot"
+    def _log_materialization_planned_event_for_asset(
+        self,
+        dagster_run: DagsterRun,
+        asset_key: AssetKey,
+        job_name: str,
+        step: "ExecutionStepSnap",
+        output: "ExecutionStepOutputSnap",
+        job_partitions_def: Optional["PartitionsDefinition"],
     ) -> None:
+        from dagster._core.definitions.partition import DynamicPartitionsDefinition
+        from dagster._core.definitions.partition_key_range import PartitionKeyRange
         from dagster._core.events import (
             AssetMaterializationPlannedData,
             DagsterEvent,
+        )
+
+        partition_tag = dagster_run.tags.get(PARTITION_NAME_TAG)
+        partition_range_start, partition_range_end = (
```

Comment:
Not blocking feedback, but pushing this down into a utility, or method on `DagsterRun` could help minimize the number of places in the codebase we need to parse these tags.

Reply:
That's a good call, though would prefer to split this out into a separate PR to fully refactor out existing references to these tags

Reply:
That makes sense